arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,867 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Visual Studio Resource Provider Client",
+    "version": "2014-04-01-preview",
+    "description": "Use these APIs to manage Visual Studio Team Services resources through the Azure Resource Manager. All task operations conform to the HTTP/1.1 protocol specification and each operation returns an x-ms-request-id header that can be used to obtain information about the request. You must make sure that requests made to these resources are secure. For more information, see https://docs.microsoft.com/en-us/rest/api/index."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/providers/Microsoft.VisualStudio/operations": {
+      "get": {
+        "tags": [
+          "accountResourceOperations"
+        ],
+        "summary": "GetOperationResources",
+        "description": "",
+        "operationId": "getOperationResources_List",
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation succeeded. The response contains the possible operations on the Microsoft.VisualStudio resource provider.",
+            "schema": {
+              "$ref": "#/definitions/ResourceOperationsDescriptorWrapper"
+            }
+         }
+        },
+        "x-ms-examples": {
+            "getOperationResources_List": {
+                "$ref": "../examples/GetOperations.json"
+            }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}": {
```

Comment:
**RP namespace and resource type as template**

As mentioned in the review summary:

(1). Expressing provider namespace as template in the paths looks odd. This can directly affect sdk end user experience. For instance, to use the service, user creates an instance of client as

```
var visualStudioClient = new VisualStudioResourceProviderClient();
```
and invoke an api as.
```
visualStudioClient.getResources.List("Microsoft.VisualStudio", other-params);
```
At this point user might be thinking why should i pass the provider namespace as parameter for all API call when i created a client that is specifically defined to work with this provider.

(2). Similar to above issue, expressing resource type as template also looks strange.

a. Given that only one resource type is Account why we are not explicitly specify it in the paths? Once we make it explicit we could have better operation group. i.e. in this case we can define operationId as `accounts_ListByResourceGroup`, then the SDK usage looks like:

```
visualStudioClient.Accounts.ListByResourceGroup(..)
```

instead of 

```
visualStudioClient.getResources.List(...);
```
b. resourceType template will be a problem when we move to azure profiles, because the profile contains static definitions of resource type along with api version. See example of profile here https://github.com/devigned/profile-playground.